app/services/articles/feeds/weighted_query_strategy.rb

```diff
@@ -0,0 +1,398 @@
+module Articles
+  module Feeds
+    # @api private
+    #
+    # This is an experimental object that we're refining to be a
+    # competetor to the existing feed strategies.
+    #
+    # @note One possible short-coming is that the query does not
+    #   account for the Forem's administrators.
+    # @note For those considering extending this, be very mindful of
+    #   SQL injection.
+    class WeightedQueryStrategy
+      # This constant defines the allowable relevance scoring methods.
+      #
+      # A scoring method should be a SQL fragement that produces a
+      # value between 0 and 1.  The closer the value is to 1, the more
+      # relevant the article is for the given user.  Note: the values
+      # are multiplicative.  Make sure to consider if you want a 0
```

Comment:
> Make sure to consider if you want a 0 multiplier for your score.

Maybe we want (0, 1] instead of [0, 1]?

Reply:
I think (0,1] is definitely a good aspiration.  Will update guidance accordingly.  (Note: not going to create a validator of this aspiration)